In [133]:
import pandas as pd
import numpy as np

In [134]:
df = pd.read_csv('formspring_data.csv', delimiter='\t')
df.shape
df.head()

(12773, 14)

userid                                               post  \
0  aguitarplayer94  Q: what&#039;s your favorite song? :D<br>A: I ...   
1      aprilpooh15                     Q: <3<br>A: </3 ? haha jk! <33   
2      aprilpooh15  Q: &quot;hey angel  you duh sexy&quot;<br>A: R...   
3      aprilpooh15                                     Q: (:<br>A: ;(   
4      aprilpooh15  Q: ******************MEOWWW*******************...   

                                                ques  \
0             what&#039;s your favorite song? :D<br>   
1                                                 <3   
2                &quot;hey angel  you duh sexy&quot;   
3                                                 (:   
4  ******************MEOWWW*************************   

                                         ans asker ans1 severity1 bully1 ans2  \
0   I like too many songs to have a favorite  None   No         0    NaN   No   
1                         </3 ? haha jk! <33  None   No         0    NaN   No   
2                   Really?!?! Thanks?! haha  None   No         0    NaN   No   
3                                         ;(  None   No         0    NaN   No   
4                                    *RAWR*?  None   No         0    NaN   No   

  severity2 bully2 ans3 severity3 bully3  
0         0    NaN   No         0    NaN  
1         0    NaN   No         0    NaN  
2         0    NaN   No         0    NaN  
3         0    NaN   No         0    NaN  
4         0    NaN   No         0    NaN

In [135]:
# copy df_num from original data
df_num = df.copy()
# remove na where col names:
col_name = ['post', 'ques', 'ans']
df_num = df_num.dropna(subset=col_name, how='any')
# replace str with int
df_num = df_num.replace({'ans1': {'Yes': 1, 'No': 0, np.nan: 0}, 'ans2': {'Yes': 1, 'No': 0, np.nan: 0}, 'ans3': {'Yes': 1, 'No': 0, np.nan: 0}})
df_num = df_num.replace({'bully1': {None: np.nan, 'None': np.nan}, 'bully2': {None: np.nan, 'None': np.nan}, 'bully3': {None: np.nan, 'None': np.nan}})
df_num = df_num.replace({'severity1': {'`0':0, '0`':0, 'o':0, 'n/a0': 0, 'N/a': 0, 'n/a0': 0, 'None': 0, np.nan: 0}, 'severity2': {'`0':0, '0`':0, 'o':0, 'n/a0': 0, 'N/a': 0, 'n/a0': 0, 'None': 0, np.nan: 0}, 'severity3': {'`0':0, '0`':0, 'o':0, 'n/a0': 0, 'N/a': 0, 'n/a0': 0, 'None': 0, np.nan: 0}})
# severity as int
df_num = df_num.astype({'severity1': 'int', 'severity2': 'int', 'severity3': 'int'})
# reset the index
# ind = np.where(((df_num.ans1 == df_num.ans2) & (df_num.ans2 == df_num.ans3) & (df_num.ans3 == df_num.ans1)))[0]
# ind = (df.ans1 == df.ans2) & (df.ans2 == df.ans3) & (df.ans1 == df.ans3)
# df_num = df_num[ind]
df_num = df_num.reset_index(drop=True)
df_num.head()

userid                                               post  \
0  aguitarplayer94  Q: what&#039;s your favorite song? :D<br>A: I ...   
1      aprilpooh15                     Q: <3<br>A: </3 ? haha jk! <33   
2      aprilpooh15  Q: &quot;hey angel  you duh sexy&quot;<br>A: R...   
3      aprilpooh15                                     Q: (:<br>A: ;(   
4      aprilpooh15  Q: ******************MEOWWW*******************...   

                                                ques  \
0             what&#039;s your favorite song? :D<br>   
1                                                 <3   
2                &quot;hey angel  you duh sexy&quot;   
3                                                 (:   
4  ******************MEOWWW*************************   

                                         ans asker  ans1  severity1 bully1  \
0   I like too many songs to have a favorite  None     0          0    NaN   
1                         </3 ? haha jk! <33  None     0          0    NaN   
2                   Really?!?! Thanks?! haha  None     0          0    NaN   
3                                         ;(  None     0          0    NaN   
4                                    *RAWR*?  None     0          0    NaN   

   ans2  severity2 bully2  ans3  severity3 bully3  
0     0          0    NaN     0          0    NaN  
1     0          0    NaN     0          0    NaN  
2     0          0    NaN     0          0    NaN  
3     0          0    NaN     0          0    NaN  
4     0          0    NaN     0          0    NaN

In [136]:
# overall score & severity
df_num['overall_ans'] = df_num.ans1 + df_num.ans2 + df_num.ans3
# df_num['overall_ans'] = df_num.ans1 + df_num.ans2 + df_num.ans3 - 1 # two agrees upon
df_num['overall_ans'][df_num.overall_ans > 1] = 1
df_num['overall_severity'] = (df_num['severity1'] + df_num['severity2'] + df_num['severity3'])/3

In [137]:
# percentage of positive observation (bullying content)
sum(df_num.overall_ans == 1)/len(df_num)

0.15155075187969924

In [138]:
# write to new dataset
df_num.to_csv('clean_formspring.csv')

<h4>Summary of dataset:</h4>
<ul>
<li>Percentage of positive variable over all data is 15%</li>
</ul>